packages/core/plugins/core-plugins-server-internal/src/discovery/plugins_discovery.ts

```diff
@@ -64,10 +66,43 @@ export function discover({
       return typeof pluginPathOrError === 'string'
         ? createPlugin$(pluginPathOrError, log, coreContext, instanceInfo, nodeInfo)
         : [pluginPathOrError];
-    }),
-    shareReplay()
+    })
   );
 
+  const pluginPkgDiscovery$ = from(coreContext.env.repoPackages ?? EMPTY).pipe(
+    filter(
+      getPluginPackagesFilter({
+        oss: coreContext.env.cliArgs.oss,
+        examples: coreContext.env.cliArgs.runExamples,
+        paths: config.additionalPluginPaths,
+        parentDirs: config.pluginSearchPaths,
+      })
+    ),
```

Comment:
This is a fundamental change. [The associated test file](https://github.com/elastic/kibana/blob/afb09ccf8a61d610e8e3d8beb2c80f413f1b33c5/packages/core/plugins/core-plugins-server-internal/src/discovery/plugins_discovery.test.ts) will have to be adapted accordingly.

Reply:
See [`6e7fcff` (#149370)](https://github.com/elastic/kibana/pull/149370/commits/6e7fcffdb87b13f672a35fd0bb27ee1be238b07d)